pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java

```diff
@@ -770,14 +770,15 @@ public static ByteBuf newActiveConsumerChange(long consumerId, boolean isActive)
         return res;
     }
 
-    public static ByteBuf newSeek(long consumerId, long requestId, long ledgerId, long entryId) {
+    public static ByteBuf newSeek(long consumerId, long requestId, long ledgerId, long entryId, int batchIndex) {
         CommandSeek.Builder seekBuilder = CommandSeek.newBuilder();
         seekBuilder.setConsumerId(consumerId);
         seekBuilder.setRequestId(requestId);
 
         MessageIdData.Builder messageIdBuilder = MessageIdData.newBuilder();
         messageIdBuilder.setLedgerId(ledgerId);
         messageIdBuilder.setEntryId(entryId);
+        messageIdBuilder.setBatchIndex(batchIndex);
```

Comment:
The `batchIndex` of the MessageIdData is an optional field, we shouldn't always use '0' here. It better to change the params type of the `batchIndex` to `Integer`. If the the `batchIndex` no a null value, set the value for the `MessageIdBuilder`